### PR TITLE
Update several dependencies

### PR DESCRIPTION
**Pull request by zalegrala**
Signed-off-by: Zach Leslie <zach.leslie@grafana.com>

**What this PR does**:

This PR updates several depenencies as part of an update for `weaveworks/common` to allow TLS specifications.

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -29,68 +29,68 @@ require (
...
 	github.com/segmentio/parquet-go v0.0.0-20221019213850-8b6e2b21a101
-	github.com/sirupsen/logrus v1.8.1
-	github.com/sony/gobreaker v0.4.1
+	github.com/sirupsen/logrus v1.9.0
+	github.com/sony/gobreaker v0.5.0
 	github.com/spf13/viper v1.12.0
-	github.com/stretchr/testify v1.8.0
-	github.com/thanos-io/thanos v0.24.0
+	github.com/stretchr/testify v1.8.1
+	github.com/thanos-io/thanos v0.29.0-rc.0
```
Comment (zalegrala) [inline go.mod:70]: This is a big one.

Comment (zalegrala) [reply go.mod:70]: How do we feel about an rc here?  We could wait, or take the rc and quickly update when 29.0 comes.

Comment (mdisibio) [reply go.mod:70]: What is the motivation for upgrading to `0.29.0-rc.0` vs the latest stable which I assume is `0.28.0`?

Comment (zalegrala) [reply go.mod:70]: There was a recent breaking change in one of the dependencies that a bunch of projects share.

https://github.com/thanos-io/thanos/blob/v0.28.1/go.mod#L19

This caused a bunch of headache for updates, which has been updated in 29.0-rc.0 to point to the new location.

**Comment by zalegrala** [comment]
Let's hold on this.  I might be having local issues that prevented me from making a smaller update.

**Comment by bboreham** [comment]
Note Tempo already has the TLS settings; the current depenency includes https://github.com/weaveworks/common/pull/256.
The only diff between that and latest `weaveworks/common` is in the wording of the help text.

**Comment by zalegrala** [comment]
Let's close this one out.  I've raised https://github.com/grafana/tempo/pull/1857 as a replacement update.

**Pull request closed without merging**